wire-runtime/src/commonMain/kotlin/com/squareup/wire/ProtoReader.kt

```diff
@@ -47,7 +47,17 @@ import okio.IOException
 /**
  * Reads and decodes protocol message fields.
  */
-class ProtoReader(private val source: BufferedSource) {
+open class ProtoReader(private val source: BufferedSource) {
```

Comment:
Would it simplify the design if we introduced a new interface, e.g. `IProtoReader<Tag>`, and have both `ProtoReader` and `ProtoReader32` implement it? Or that would break backwards compat?

Reply:
I would love it, but the generic `Tag` would be boxed on every use, potentially at significant runtime cost.